CMakeLists.txt

```diff
@@ -918,8 +918,8 @@ if(CMAKE_COMPILER_IS_GNUCXX AND HPX_WITH_GCC_VERSION_CHECK)
 endif()
 
 if(MSVC)
-  if(NOT (MSVC11 OR MSVC12 OR MSVC14))
-    hpx_error("MSVC x64 2012 or higher is required.")
+  if(NOT (MSVC14))
```

Comment:
Looks good, however its inconvenient to have a CMake variable for each MSVC version, a better approach would be the usage of the `MSVC_VERSION` variable: `if(NOT (MSVC_VERSION LESS 1900))`